src/views/Validator/Validator.hooks.ts

```diff
@@ -2,7 +2,8 @@ import { useLocation, useParams } from 'react-router-dom'
 
 export function useScriptPath() {
   const { fileName } = useParams()
-  const { state } = useLocation()
+  // TODO(router): useLocation is not type-safe. Refactor this route to avoid using it.
+  const { state } = useLocation() as { state: { externalScriptPath: string } }
```

Comment:
@allansson now I see what you meant, and I barely scratched the surface...

Reply:
The same goes for getting route params via `useParams` or search params via `useSearchParams`. You're just casting things hoping that your routes are structured in a certain way.